search-app/src/cmr/search/services/query_walkers/facet_condition_resolver.clj

```diff
@@ -4,10 +4,24 @@
   (:require
    [cmr.common-app.services.search.group-query-conditions :as gc]
    [cmr.common-app.services.search.query-model :as cqm]
-   [cmr.common.util :as util]))
+   [cmr.common.util :as util]
+   [cmr.search.services.query-execution.facets.facets-v2-results-feature :as fvrf]))
+
+(defn- string-condition-for-v2-facet-field?
+  "Returns true if the given string condition is on the given v2 facet field"
+  [c field-key]
+  (if (or
+       ;; The first check handles the data-center-h field since its query condition field
+       ;; :organization.humanized2.value does not match the regex on the second check
+       (= (str (fvrf/facets-v2-params->elastic-fields field-key) ".value") (str (:field c)))
```

Comment:
This code is difficult for me to understand.

I guess I'm not sure what `c` looks like. Where is that .value coming from?
So `c` will look like:
`{:field :science-keywords.humanized.value ...}`?
Why do we also do the `re-matches` check below? Ideally the first check should work for all of the fields.